csync/src/std/c_utf8.cc

```diff
@@ -64,26 +60,33 @@ char* c_utf8_from_locale(const mbchar_t *wstr)
     WideCharToMultiByte(CP_UTF8, 0, wstr, len, mdst, size_needed, NULL, NULL);
     dst = mdst;
   }
+  return dst;
 #else
-  dst = c_strdup(wstr);
+    QTextDecoder dec(QTextCodec::codecForLocale());
+    QString s = dec.toUnicode(wstr, qstrlen(wstr));
```

Comment:
`const QString s = ....` ?

Tested on Linux and works, but might be an issue on MacOSX